cl/citations/tests.py

```diff
@@ -355,39 +429,350 @@ def test_disambiguate_citations(self):
+            (u'asdf.’ ” 123 \n U.S., at 456. Foo bar foobar',
+             '<pre class="inline">asdf.’ </pre><span class="citation no-link">'
+             '<span class="antecedent">” </span><span class="volume">123'
+             '</span> \n <span class="reporter">U.S.</span>, at <span class='
+             '"page">456</span></span><pre class="inline">. Foo bar foobar'
+             '</pre>'),
+
+            # First kind of supra citation (standard kind)
+            ('asdf, supra, at 2. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf,</span><span> supra</span><span>, at '
+             '</span><span class="page">2</span></span><pre class="inline">'
+             '. foobar</pre>'),
+
+            # Second kind of supra citation (with volume)
+            ('asdf, 123 supra, at 2. foo bar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf,</span> <span class="volume">123</span>'
+             '<span> supra</span><span>, at </span><span class="page">2</span>'
+             '</span><pre class="inline">. foo bar</pre>'),
+
+            # Third kind of supra citation (sans page)
+            ('asdf, supra, foo bar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf,</span><span> supra</span></span><pre '
+             'class="inline">, foo bar</pre>'),
+
+            # Fourth kind of supra citation (with period)
+            ('asdf, supra. foo bar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf,</span><span> supra</span></span><pre '
+             'class="inline">. foo bar</pre>'),
+
+            # Supra citation across line break
+            ('asdf, supra, at\n99 (quoting foo)',
+             '<pre class="inline"></pre><span class="citation no-link"><span'
+             ' class="antecedent">asdf,</span><span> supra</span><span>, at\n'
+             '</span><span class="page">99</span> </span><pre class="inline">'
+             '(quoting foo)</pre>'),
+
+            # First kind of id. citation ("Id., at 123")
+            ('asdf, id., at 123. Lorem ipsum dolor sit amet',
+             '<pre class="inline">asdf, </pre><span class="citation no-link">'
+             'id.,<span class="after_tokens"> <span class="after_token">at'
+             '</span> <span class="after_token">123.</span> <span class="'
+             'after_token">Lorem</span> </span></span><pre class="inline">'
+             'ipsum dolor sit amet</pre>'),
+
+            # Second kind of id. citation ("Ibid.")
+            ('asdf, Ibid. Lorem ipsum dolor sit amet',
+             '<pre class="inline">asdf, </pre><span class="citation no-link">'
+             'Ibid.<span class="after_tokens"> <span class="after_token">Lorem'
+             '</span> <span class="after_token">ipsum</span> <span class="'
```

Comment:
Is there a reason to put each of the `after_token`s in their own `span`?

Reply:
No, this is sort of insane. Will fix

Reply:
Fixed in b1e76cd